scripts/redeploy.py

```diff
@@ -107,21 +107,35 @@ def on_receive_jenkins_poke():
     )
 
     print("Retrieving .tar.gz file: %s" % tar_gz_url)
+
+    # we rely on the fact that flask only serves one request at a time to
+    # ensure that we do not overwrite a tarball from a concurrent request.
     filename = download_file(tar_gz_url)
     print("Downloaded file: %s" % filename)
-    name_str = filename.replace(".tar.gz", "")
-    untar_to(filename, arg_extract_path)
 
-    extracted_dir = os.path.join(arg_extract_path, name_str)
-
-    if arg_should_clean:
-        os.remove(filename)
+    try:
+        # we extract into a directory based on the build number. This avoids the
+        # problem of multiple builds building the same git version and thus having
```

Comment:
it *should*, but:

(a) the tarball from jenkins might be corrupted, so we replace a good deploy with a bad one
(b) we'll be overwriting the live deployment, which means people might see half-written files.